src/inmanta/util/__init__.py

```diff
@@ -863,3 +871,118 @@ def check_for_pool_exhaustion(self) -> None:
 
     def _reset_counter(self) -> None:
         self._exhausted_pool_events_count = 0
+
+
+def remove_comment_part(to_clean: str) -> str:
+    """
+    Remove the comment part of a given string and ensure that the length of the string is greater than 0
+
+    :param to_clean: The string to clean
+    :return: A cleaned string
+    """
+    # Refer to PEP 508. A requirement could contain a hashtag
+    to_clean = to_clean.strip()
+    if to_clean.startswith("#"):
+        raise ValueError("The name of the requirement cannot be a comment!")
```

Comment:
Why can't we have lines with only comments, as described in the pep?

This method should document when it raises an exception, because now it is super confusing.

Also, using the same exception for errors and comments is very weird to me.

Reply:
I moved the raising of the error to the caller: makes more sense and was probably creating more confusion than necessary

I agree that different exception errors should be used, even though in those particular cases, they could be considered a ValueError.